modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/extension/GlobalElementComponentModelModelProperty.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.dsl.model.extension;
+
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.config.spring.dsl.model.ComponentModel;
+
+import java.util.List;
+
+/**
+ * Property to store all Mule global elements that are contained in an extension written in XML.
+ *
+ * @since 4.0
+ */
+public class GlobalElementComponentModelModelProperty implements ModelProperty {
```

Comment:
If some classes are for Java extensions and others for XML extensions I would expect extension.java and extension.xml